stock_location_tank/models/stock_location.py

```diff
@@ -0,0 +1,169 @@
+                        " and it has a manufacturing order assigned."
+                    ),
+                }
+            )
+
+    @api.depends("production_id")
+    def _compute_is_blocked(self):
+        for rec in self:
+            rec.is_blocked = True if rec.is_tank and rec.production_id else False
+
+    @api.depends("usage", "is_tank", "quant_ids.product_id.stock_move_ids")
+    def _compute_production_id(self):
+        for rec in self:
+            # Mirar els estats to_close, confirmed, progress
+            # (son els unics valids per el nostre cas)
+            if rec.is_tank and rec.usage != "view":
+                production = rec.quant_ids.product_id.stock_move_ids.filtered(
+                    lambda x: x.state not in ("draft", "cancel", "done")
+                ).production_id
+                if len(production) > 1:
+                    raise UserError(
+                        _(
+                            "LOCATION: The location %s is blocked. Probably you need to"
+                            " review the pending manufacturing orders related to this location"
+                        )
+                        % rec.name
+                    )
+                rec.production_id = production
+            else:
+                rec.production_id = False
+
+    @api.depends("quant_ids.quantity")
+    def _compute_capacity_occupied(self):
+        for rec in self:
+            rec.capacity_occupied = sum(rec.quant_ids.mapped("quantity"))
+
+    @api.depends("capacity_occupied")
+    def _compute_percentage_occupied(self):
+        for rec in self:
+            if rec.capacity <= 0:
+                rec.percentage_occupied = 0
+            else:
+                rec.percentage_occupied = rec.capacity_occupied / rec.capacity * 100
+
+    def action_view_mrp_production(self):
+        self.ensure_one()
+        return {
+            "name": _("Manufacturing Orders"),
+            "type": "ir.actions.act_window",
+            "view_mode": "form",
+            "res_model": "mrp.production",
+            "res_id": self.production_id.id,
+        }
+
+    @api.constrains("allowed_uom_id")
+    def _check_allowed_uom_id(self):
+        for rec in self:
+            if rec.quant_ids.filtered(
+                lambda x: x.product_uom_id != rec.allowed_uom_id and x.quantity > 0
+            ):
+                raise UserError(
+                    _("You have stock movements with different unit of measure")
+                )
+
+    @api.constrains("is_tank")
+    def _check_is_tank(self):
+        for rec in self:
+            if rec.quant_ids.filtered(lambda x: x.quantity > 0):
+                raise UserError(_("You can't change the type of location with stock"))
+            if not rec.is_tank:
+                rec.sequence_id = False
+                rec.allowed_product_ids = False
+                rec.capacity = 0
+                rec.allowed_uom_id = False
```

Comment:
We should not assign nothing in a constraint, if we need to do that better inheriting `create` and/or `write`